### ISSUE TITLE
a11y: Rating reads as Unticked even when ticked

### ISSUE THREAD

**Issue by pearlyong** [opengovsg/FormSG#5503]
**Checked**
- Mac Voiceover reads Required Selected Invalid Data even when selected
- iPhone Voiceover whether it is selected or not, it reads unticked, reads 'Radio button unticked, 4 of 5 required' when selected 'Radio button unticked, 3 of 5 required'
- Talkback reads Option xx Ticked/Unticked Option 2 of 5 Radio Button

**Proposal**
- Could we [implement the scale](https://www.figma.com/file/9oE4ECjdeZQ1RBpwZVtnB3/Form-Design-Master?node-id=14439%3A138843&t=cNKitpnP9oUFpWBf-4)? And can we read out the scale before the stars, even though the placement is the other way around?
- On MacOS Voiceover, when it's selected, can we not read out Invalid data on Voiceover?
- On iOS Voiceover, can we read out what the user selects when selected? (e.g. selected, 2 of 5)

**Comment by staceytan1998** [comment]
**Suggestion from Etch empathy:**

Add an accessible label for each option using ARIA, such as:

1 out of 5 stars, 2 out of 5 stars, …, 5 out of 5 stars

The word 'Stars' should be added both to describe what is being visually represented and to help implicitly communicate the scale of the rating system. For Heart rating, we can use 'Heart'! 

**Screenshot**
https://drive.google.com/file/d/1VoAoqBrHu1a8kG49cBh4So1JqwdFvf8X/view?usp=share_link

**Comment by foochifa** [comment]
For MacOS Voiceover, the reason why its reading out as required invalid data is because [aria-required](https://github.com/w3c/html-aam/issues/394) is tagged to each singular radio button. It should be tagged to the rating field container instead, like how we did it for radio fields. Not sure if this causes the 'untick' issue for iPhone Voiceover and Talkback.

On Stacey's point, we should include aria-labels for each radio, something like [this](https://w3c.github.io/aria-practices/examples/radio/radio-rating.html) from w3c

**Comment by foochifa** [comment]
PR #5767 fixes the Mac Voiceover problem. Will close this issue and open a new one for the iOS Voiceover bug
